dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/managed/ManagedFileResourceRepositoryImpl.java

```diff
@@ -0,0 +1,211 @@
+package de.digitalcollections.commons.file.backend.impl.managed;
+
+import de.digitalcollections.commons.file.backend.impl.FileResourceRepositoryImpl;
+import de.digitalcollections.model.api.identifiable.resource.FileResource;
+import de.digitalcollections.model.api.identifiable.resource.MimeType;
+import de.digitalcollections.model.api.identifiable.resource.exceptions.ResourceIOException;
+import de.digitalcollections.model.api.identifiable.resource.exceptions.ResourceNotFoundException;
+import de.digitalcollections.model.impl.identifiable.resource.FileResourceImpl;
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Objects;
+import java.util.UUID;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.StringUtils;
+
+/**
+ * A binary repository using filesystem.
+ * see http://docs.oracle.com/javase/tutorial/essential/io/fileio.html
+ * see https://docs.oracle.com/javase/tutorial/essential/io/file.html
+ * see http://michaelandrews.typepad.com/the_technical_times/2009/10/creating-a-hashed-directory-structure.html
```

Comment:
```suggestion
 * see https://medium.com/eonian-technologies/file-name-hashing-creating-a-hashed-directory-structure-eabb03aa4091
```

Link geht nicht mehr, aber es gibt anscheinend einen identischen Eintrag auf medium.com